internal/common/mqtt/mqttconnection_test.go

```diff
@@ -22,12 +22,52 @@ import (
 
 const Host = "broker.hivemq.com"
 const port = "1883"
+const InvalidHost = "invalid"
+
+func initializeTest(Host string, clientID string) {
+	InitClientData()
+	StartMQTTClient(Host, clientID)
+}
 
 func TestStartMQTTClient(t *testing.T) {
 	t.Run("Success", func(t *testing.T) {
+		initializeTest(Host, "testClient")
+		client := clientData["testClient"]
+		message := Message{"appid", "Test data for testing"}
+		client.Publish(message, "home/livingroom")
+		err := StartMQTTClient(Host, "testClient")
+		if len(err) > 0 {
+			t.Error("Unexpected err", err)
+		}
+	})
+	t.Run("Fail", func(t *testing.T) {
 		InitClientData()
-		StartMQTTClient(Host, "testClient")
-		t.Log("Client connected")
+		err := StartMQTTClient(InvalidHost, "testClientFailure")
+		if len(err) > 0 {
+			t.Log("Unexpected err", err)
+		}
+	})
+
+}
+
+func TestCheckforConnectionPass(t *testing.T) {
```

Comment:
Please revise the function name to `TestCheckforConnection`.

```suggestion
func TestCheckforConnection(t *testing.T) {
```